src/app/users/get-user.ts

```diff
@@ -1,65 +1,99 @@
-import {
-  SemanticAttributes,
-  asyncRunInSpan,
-  addAttributesToCurrentSpan,
-  ENDUSER_ALIAS,
-} from "@services/tracing"
+import { SemanticAttributes, asyncRunInSpan } from "@services/tracing"
 import { UsersRepository } from "@services/mongoose"
-import { IpFetcher } from "@services/ipfetcher"
 import { getIpConfig } from "@config/app"
+import { IpFetcher } from "@services/ipfetcher"
+import { UsersIpRepository } from "@services/mongoose/users-ip"
 import { RepositoryError } from "@domain/errors"
 
 const users = UsersRepository()
+const usersIp = UsersIpRepository()
 
 export const getUserForLogin = async ({
   userId,
   ip,
+  logger,
 }: {
-  userId: string
-  ip?: string
+  userId: UserId
+  ip?: Ip
```

Comment:
Same here - should be a primitive type as input. Convert to domain type via `checkedToIp` which can make sure it really is an ip.

Reply:
do you have an example to follow this somewhere else in the code?

Reply:
here for example - https://github.com/GaloyMoney/galoy/blob/main/src/app/wallets/get-on-chain-fee.ts#L65

Its not 100% consistent but it is what we have decided to converge to. And if a type got returned from app then that is also trusted as long as conversion happened in app layer.